sim/team-validator.ts

```diff
@@ -2468,6 +2477,8 @@ export class TeamValidator {
 					continue;
 				}
 
+				if (eggMovesOnly && learned.charAt(1) !== 'E') continue;
```

Comment:
Another approach would be to put the `checkingPrevo && !originalSpecies.prevo` check here and remove the `eggMovesOnly` variable.